src/ensembl/production/metadata/api/genome.py

```diff
@@ -80,64 +112,118 @@ def fetch_taxonomy_ids(self, taxonomy_names):
+
+        Returns:
+            List[Tuple[Genome, Organism, Assembly, EnsemblRelease]]: A list of tuples containing the fetched genome information.
+            Each tuple contains the following elements:
+                - Genome: An instance of the Genome class.
+                - Organism: An instance of the Organism class.
+                - Assembly: An instance of the Assembly class.
+                - EnsemblRelease: An instance of the EnsemblRelease class.
+
+        Notes:
+            - The parameters are not mutually exclusive, meaning more than one of them can be provided at a time.
+            - The function uses a database session to execute the query and returns the results as a list of tuples.
+            - The results are ordered by the Ensembl name.
+
+        Example usage:
+            genome_info = fetch_genomes(genome_id=12345)
+        """
+        # Parameter validation
         genome_id = check_parameter(genome_id)
         genome_uuid = check_parameter(genome_uuid)
+        organism_uuid = check_parameter(organism_uuid)
         assembly_accession = check_parameter(assembly_accession)
+        assembly_name = check_parameter(assembly_name)
         ensembl_name = check_parameter(ensembl_name)
         taxonomy_id = check_parameter(taxonomy_id)
         group = check_parameter(group)
         group_type = check_parameter(group_type)
 
+        # Construct the initial database query
         genome_select = db.select(
-            Genome, Organism, Assembly
-        ).join(Genome.assembly).join(Genome.organism)
-
+            Genome, Organism, Assembly, OrganismGroupMember, OrganismGroup
+        ).select_from(Genome) \
+            .join(Organism, Organism.organism_id == Genome.organism_id) \
+            .join(Assembly, Assembly.assembly_id == Genome.assembly_id) \
+            .join(OrganismGroupMember, OrganismGroupMember.organism_id == Organism.organism_id) \
+            .join(OrganismGroup, OrganismGroup.organism_group_id == OrganismGroupMember.organism_group_id) \
+
+        # Apply group filtering if group parameter is provided
         if group:
             group_type = group_type if group_type else ['Division']
-            genome_select = db.select(
-                Genome, Organism, Assembly, OrganismGroup
-            ).join(Genome.assembly).join(Genome.organism) \
-                .join(Organism.organism_group_members) \
-                .join(OrganismGroupMember.organism_group) \
-                .filter(OrganismGroup.type.in_(group_type)).filter(OrganismGroup.name.in_(group))
+            genome_select = genome_select.filter(OrganismGroup.type.in_(group_type)).filter(OrganismGroup.name.in_(group))
 
+        # Apply additional filters based on the provided parameters
         if unreleased_only:
+            # Question (Bilal): When adding unreleased genomes do we populate genome_releases table?
+            # if that's the case the test data doesn't do so (this will return en empty list)
```

Comment:
Unreleased genomes will not have a genome_release. Your logic is odd in my mind as I would look for genomes without a genome_release, but if it works I am fine with it.